File: day1.py

f = open('./input1.txt', 'r')

floor = 0
position = 0
enteredBasement = False
basement = 0

for c in f.read():
    position += 1
    if c == '(':
        floor += 1
    elif c == ')':
        floor -= 1
    if floor == -1 and not enteredBasement:
        enteredBasement = True
        basement = position

print(str(floor) + '\n' + 'entered basement at: ' + str(basement))

File: day1.java

import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Paths;

public class day1 {

	public static void main(String[] args) throws IOException {

		String input = new String(Files.readAllBytes(Paths.get("input1.txt")));

		int floor = 0;
		int basement = 0;
		int position = 0;

		for(char c : input.toCharArray()) {
			if(c == '(') floor++;
			if(c == ')') floor--;

			if(basement == 0 && floor == -1) basement = position + 1;

			position++;
		}

		System.out.println(floor);

		System.out.println("basement was entered at " + basement);
	}
}
